### PR TITLE
chore: Bump dependencies

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

- Bump stackable-operator to 0.67.2, and other dependencies
  - Fix imports, and format code
- Bump github action versions

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,18 +9,18 @@ version = "1.1.1"
...
-semver = "1.0"
-snafu = "0.8"
-strum = { version = "0.26", features = ["derive"] }
-tokio = { version = "1.29", features = ["full"] }
-tracing = "0.1"
-pin-project = "1.0"
-warp = "0.3"
-tar = "0.4"
-flate2 = "1.0"
+flate2 = "1.0.30"
```
Comment (sbernauer) [inline Cargo.toml]: Nit (optional): For most operators we only use the major.minor version. I would argue that this is similar to an operator but how knows :D

File: Cargo.toml [hunk 2]
```diff
@@ -9,18 +9,18 @@ version = "1.1.1"
...
+futures = { version = "0.3.30", features = ["compat"] }
+pin-project = "1.1.5"
+semver = "1.0.23"
+snafu = "0.8.2"
+strum = { version = "0.26.2", features = ["derive"] }
+tar = "0.4.40"
+tokio = { version = "1.37.0", features = ["full"] }
```
Comment (sbernauer) [inline Cargo.toml]: Nit (optional): It would be great if we could reduce the feature-set

**Review by sbernauer — DISMISSED** [review]
Only nits

File: Cargo.toml (hunk 2 above)
Comment (NickLarsenNZ) [reply Cargo.toml]: I thought about that too, but I think it can Perhaps be done in a separate PR.

Comment (sbernauer) [reply Cargo.toml]: Alright

File: Cargo.toml (hunk 1 above)
Comment (NickLarsenNZ) [reply Cargo.toml]: Done in 2ee2e02

**Review by sbernauer — APPROVED** [review]

**Comment by NickLarsenNZ** [comment]
Tagged

**Pull request merged**